src/shared/utils/keyEscapeUtils.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule keyEscapeUtils
+ */
+
+'use strict';
+
+function escape(key) {
+  var escapeRegex = /[=:]/g;
+  var escaperLookup = {
+    '=': '=0',
+    ':': '=2',
+  };
+  var escapedString = ('' + key).replace(
+    escapeRegex,
+    function(match) {
+      return escaperLookup[match];
+    }
+  );
+
+  return '$' + escapedString;
+}
+
+function unescape(key) {
+  var unescapeRegex = /(=0|=2)/g;
+  var unescaperLookup = {
+    '=0': '=',
+    '=2': ':',
+  };
+  var keySubstring = (key[0] === '.') ? key.substring(2) : key.substring(1);
```

Comment:
Can you explain this line?  Maybe I'm just tired and not understanding, but I would have expected some sort of a check for a '$' symbol.